p2p/client_test.go

```diff
@@ -20,16 +21,37 @@ import (
 
 func TestClientStartup(t *testing.T) {
 	privKey, _, _ := crypto.GenerateEd25519Key(rand.Reader)
-	client, err := NewClient(config.P2PConfig{}, privKey, "TestChain", test.NewLogger(t))
+	gater, err := conngater.NewBasicConnectionGater(nil)
 	assert := assert.New(t)
 	assert.NoError(err)
-	assert.NotNil(client)
+	testCases := []config.P2PConfig{
+		{},
+		{
+			ListenAddress: "",
+			Seeds:         "",
+			BlockedPeers:  "",
+			AllowedPeers:  "/ip4/127.0.0.1/tcp/7676/p2p/12D3KooWM1NFkZozoatQi3JvFE57eBaX56mNgBA68Lk5MTPxBE4U",
+			// unblocking a not previously explicitly blocked peer shouldn't result in an error
+		},
+		{
+			ListenAddress: "",
+			Seeds:         "",
+			BlockedPeers:  "/ip4/127.0.0.1/tcp/7676/p2p/12D3KooWM1NFkZozoatQi3JvFE57eBaX56mNgBA68Lk5MTPxBE4U",
+			AllowedPeers:  "",
+		},
+	}
 
-	err = client.Start(context.Background())
-	defer func() {
-		_ = client.Close()
-	}()
-	assert.NoError(err)
+	for _, testCase := range testCases {
+		client, err := NewClient(testCase, privKey, "TestChain", gater, test.NewLogger(t))
```

Comment:
The test is tabular, which is cool, but gater is reused, so cases depends on each other, which is not.
Also, please use subtests (see: https://mj-go.in/golang/table-driven-tests-in-go#run-table-driven-tests-as-subtests)